--- ./src/astrobee/mobility/choreographer/src/validator.cc	2024-01-10 16:34:47.790682249 +0900
+++ ./src/astrobee/mobility/choreographer/src/validator.cc.new	2024-01-10 16:34:47.794682249 +0900
@@ -270,7 +270,7 @@ void Validator::PublishMarkers() {
   for (it = zones_.zones.begin(); it != zones_.zones.end(); it++) {
     switch (it->type) {
     case ff_msgs::Zone::KEEPOUT:
-      marker.color.a = 0.1;
+      marker.color.a = 0.3;
       marker.color.r = 1.0;
       marker.color.g = 0.0;
       marker.color.b = 0.0;
