--- build_install_debians.sh	2024-01-05 18:38:58.430289482 +0900
+++ build_install_debians.sh.new	2024-01-05 18:42:35.146281515 +0900
@@ -25,7 +25,7 @@ debian_loc=$(dirname "$(readlink -f "$0"
 dist=$(. /etc/os-release && echo $UBUNTU_CODENAME)
 build_list=()
 
-sudo apt-get install -y devscripts equivs libproj-dev
+sudo -E apt-get install -y devscripts equivs libproj-dev
 # delete old files (-f avoids 'no such file' warning on first run)
 rm -f *.deb *.debian.tar.xz *.orig.tar.gz *.dsc *.build *.buildinfo *.changes *.ddeb
 
@@ -39,12 +39,12 @@ case $dist in
   bionic)
     echo "Ubuntu 18 detected"
      # jps3d deps
-    sudo apt-get install -y libvtk6.3 libboost-filesystem1.62.0 libboost-system1.62.0
+    sudo -E apt-get install -y libvtk6.3 libboost-filesystem1.62.0 libboost-system1.62.0
     ;;
   focal)
     echo "Ubuntu 20 detected"
     #jps3d deps
-    sudo apt-get install -y libvtk7.1p libboost-filesystem1.71.0 libboost-system1.71.0
+    sudo -E apt-get install -y libvtk7.1p libboost-filesystem1.71.0 libboost-system1.71.0
     ;;
   *)
     echo "No supported distribution detected"
@@ -62,8 +62,8 @@ export DEBEMAIL="nospam@nospam.org"
 for pkg in ${build_list[@]}
 do
   cd ${debian_loc}/$pkg &&
-  sudo mk-build-deps -i -r -t "apt-get --no-install-recommends -y" control &&
+  sudo -E mk-build-deps -i -r -t "apt-get --no-install-recommends -y" control &&
   cd ${debian_loc} &&
   ./build_${pkg}.sh &&
-  sudo dpkg -i *${pkg}*.deb || exit 1
+  sudo -E dpkg -i *${pkg}*.deb || exit 1
 done
